pkg/ebpf/maps/generic_map_test.go

```diff
@@ -572,3 +572,25 @@ func TestIterateWithPointerKey(t *testing.T) {
 	require.NoError(t, it.Err())
 	require.Equal(t, expectedNumbers, actualNumbers)
 }
+
+func TestGenericHashMapCanUseBatchAPI(t *testing.T) {
+	if !BatchAPISupported() {
+		t.Skip("Skipping because batch API not supported")
+	}
+
+	hash, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type:       ebpf.Hash,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 10,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
require.NoError(t, err)

Reply:
If we cannot create a map, then that is not really an error scenario. It just means we cannot carry out our test

Reply:
`require.Error` terminates the test if the condition is not met

Reply:
Yes it would behave the same.

But using `t.Fatal` makes it more clear that failing to create a hash map is not an error condition for this test, but a fatal error in the setup.

Reply:
I think `require.NoError` will do the same, i.e. call `t.Fatal`

Reply:
I guess it calls `t.FailNow`, but that is just a log followed by `t.Fatal`.

Reply:
I just thought using `t.Fatal` would make it easier to see for someone reading the code that the setup failed, and not the test itself.

But I guess it is not as clear as I thought. I will change to use NoError.